ui/perfherder/helpers.js

```diff
@@ -429,25 +429,53 @@ export const addResultsLink = (taskId) => {
   return `${taskLink}${taskId}${resultsPath}`;
 };
 
-export const getTextualSummary = (
-  alerts,
-  alertSummary,
-  copySummary = null,
-  alertsWithVideos = [],
-) => {
-  let resultStr = '';
-  const improved = sortBy(
-    alerts.filter((alert) => !alert.is_regression),
-    'amount_pct',
-  ).reverse();
-  const regressed = sortBy(
-    alerts.filter(
-      (alert) => alert.is_regression && alert.status !== alertStatusMap.invalid,
-    ),
-    'amount_pct',
-  ).reverse();
+export class TextualSummary {
+  constructor(alerts, alertSummary, copySummary = null, alertsWithVideos = []) {
+    this.alerts = alerts;
+    this.alertSummary = alertSummary;
+    this.copySummary = copySummary;
+    this.alertsWithVideos = alertsWithVideos;
+    this.resultStr = '';
```

Comment:
I don't think we should store this variable at instance level. For example, calling `markdown` property multiple times would simply duplicate its value. Which is weird.